### PR TITLE
feat(ui): Glossary Terms table on TableV2 — adaptable search, column resize, drag reorder, sticky header

### PR DESCRIPTION
**Pull request by siddhant1**
## What

Completes the migration of the **Glossary Terms** table to the React Aria `TableV2`/`TableCard` stack and restores the behaviors that were lost during the migration.

### Changes
- **Adaptable search bar** — the toolbar search input now grows/shrinks to fill available width, while Status / Bulk Edit / Expand-All stay fixed-size and always visible (no more clipped buttons on narrow viewports).
- **Column resize indicator** — the `ColumnResizer` handle was invisible under Tailwind v4 (deprecated `bg-[--var]` arbitrary-value syntax silently dropped). Switched to the valid `bg-border-secondary` / `bg-border-brand` utilities: a subtle divider at idle, brand-blue while dragging.
- **Row drag-and-drop reorder / re-parent** — React Aria's `Row` strips native HTML5 drag props (`filterDOMProps`), so the old `onRow` drag handlers never reached the DOM. Reimplemented with `useDragAndDrop` (`dragAndDropHooks`), reusing the existing move-confirmation modal + PATCH flow. Load-more placeholder rows are excluded as drag sources/targets.
- **Sticky header** — the table now owns its own vertical scroll (`scroll.y`) so the sticky `<thead>` engages; the outer container no longer scrolls; and infinite scroll is re-pointed at the internal scroll region.

### Files
- `GlossaryTermTab.component.tsx` — search layout, RAC drag-and-drop, internal-scroll + sticky-header wiring, infinite-scroll rework.
- `common/Table/TableV2.tsx` — column resize indicator token fix.

## Testing
Manual verification in the Glossary → Terms tab:
- Resize the viewport → search adapts, buttons stay visible.
- Drag a column edge → visible resize handle; drag a row onto another → re-parent modal → nests; drop on empty area → moves to top level.
- Scroll the term list → header stays pinned; infinite scroll still loads more (regular + search); short lists auto-fill.

> Note: the `scroll.y` offset (`calc(100vh - 350px)`) may need minor tuning per layout.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

----
## Summary by Gitar

- **Drag-and-drop mechanics:**
  - Added `useDragAndDrop` from `react-aria-components` to handle row re-parenting and promotion, replacing deprecated HTML5 handlers.
  - Added `getDropOperation` logic to manage valid drop targets and prevent interaction with load-more rows.
- **Component architecture:**
  - Moved table layout to `TableCard.Root` and updated `GlossaryTermTab` to use internal `scroll.y` for consistent sticky header behavior.
- **Refactoring:**
  - Removed `useInView` infinite scroll observer in favor of the table's internal scroll management.
  - Added `glossaryTermByFqn` lookup memo for efficient drag-and-drop target identification.
- **Bug fixes:**
  - Fixed missing `permissions.EditAll` dependency in `extraTableFilters` memo to ensure accurate bulk-edit button visibility.

<sub>This will update automatically on new commits.</sub>

<details open><summary><h3>Greptile Summary</h3></summary>

This PR completes the Glossary Terms table migration to the React Aria `TableV2`/`TableCard` stack, restoring four behaviors lost during the migration: an adaptable search bar, a visible column-resize indicator, row drag-and-drop reorder/re-parent via `useDragAndDrop`, and a sticky header backed by an internal scroll region. Playwright tests are updated to match the new DOM structure — `data-testid` selectors replace Ant Design class-name selectors, and scroll helpers walk the DOM to find the real scrollable container instead of the now-static outer div.

- **`GlossaryTermTab.component.tsx`** — switches from `useInView`-based infinite scroll to a `MutationObserver` + scroll-event hybrid, wires up React Aria drag-and-drop with `useDragAndDrop`, and routes internal scroll through the TableV2 `scroll.y` prop so the sticky header engages.
- **`TableV2.tsx`** — fixes the `ColumnResizer` color token from the invalid `bg-[--var]` arbitrary-value syntax to the valid `bg-border-secondary` / `bg-border-brand` Tailwind v4 utilities.
- **Test files** — `LargeGlossaryPerformance.spec.ts` and `GlossaryStatusFilterLargeDataset.spec.ts` now use the DOM-walking helper to locate the real scroll container, resolving the silent-discard scroll tests flagged in earlier review rounds.
</details>

<details open><summary><h3>Confidence Score: 5/5</h3></summary>

Safe to merge; all changes are additive UI behavior or test selector updates with no data-model or API contract changes.

The core drag-and-drop and infinite-scroll rewrites are mechanically sound and the previously flagged scroll-container issues in the test suite have been addressed. The two newly found concerns are narrow edge cases that do not affect primary user flows and are mitigated by surrounding guards.

GlossaryTermTab.component.tsx carries the most complex logic (dual MutationObserver + scroll handler, drag-and-drop wiring, glossaryTermByFqn map) and warrants the closest reading; DraggableMenuItemV2.component.tsx should be verified for keyboard accessibility before the next accessibility audit.
</details>

<details open><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| openmetadata-ui/src/main/resources/ui/src/components/Glossary/GlossaryTermTab/GlossaryTermTab.component.tsx | Major refactor: adds React Aria drag-and-drop, internal table scroll, infinite-scroll rework, and adaptable search toolbar; several subtle issues in scroll/drag logic carry over from prior review cycles. |
| openmetadata-ui/src/main/resources/ui/src/components/common/Table/TableV2.tsx | Fixes ColumnResizer color token from invalid arbitrary-value syntax to valid Tailwind v4 utilities; adds data-testid to toolbar and column dropdown title; straightforward change. |
| openmetadata-ui/src/main/resources/ui/src/components/common/Table/DraggableMenu/DraggableMenuItemV2.component.tsx | Restructures drag and toggle responsibilities: full row becomes drag source, nested button handles visibility toggle; removes onPress from Dropdown.Item, creating a keyboard accessibility gap. |
| openmetadata-ui/src/main/resources/ui/src/components/Glossary/glossaryV1.less | Replaces old Ant Design-targeted sticky-header CSS with RAC-compatible drag row styles and top-level drop-target highlight; clean change. |
| openmetadata-ui/src/main/resources/ui/playwright/utils/glossary.ts | Updates test utilities to use data-testid selectors, scopes the Move modal button to the confirmation modal, and adds helper ensureColumnsVisible; changes look correct. |
| openmetadata-ui/src/main/resources/ui/playwright/e2e/Features/Glossary/LargeGlossaryPerformance.spec.ts | Extracts shared helpers to walk the DOM for the real scroll container; updates row selectors from Ant Design class names to data-row-key; comprehensive fix for the scroll-to-load path. |
| openmetadata-ui/src/main/resources/ui/playwright/e2e/Features/Glossary/GlossaryStatusFilterLargeDataset.spec.ts | Adopts the same DOM-walking scroll helper and data-testid selectors for the status-filter infinite scroll tests; fixes the silent-discard problem flagged in a previous review. |
| openmetadata-ui/src/main/resources/ui/playwright/e2e/Features/Glossary/GlossaryHierarchy.spec.ts | Adds new H-M03b test: drag a nested term to root level; uses existing dragAndDropTerm/confirmationDragAndDropGlossary helpers correctly. |

</details>

<sub>Reviews (22): Last reviewed commit: ["Merge branch &#39;main&#39; into glossary-table-..."](https://github.com/open-metadata/openmetadata/commit/ced1580fab49ca82d0f64eaf5b531621a4660bdc) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=41603523)</sub>

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/src/components/Glossary/GlossaryTermTab/GlossaryTermTab.component.tsx [hunk 1]
```diff
@@ -1502,6 +1508,138 @@ const GlossaryTermTab = ({ isGlossary, className }: GlossaryTermTabProps) => {
...
+      let operation: DropOperation = 'move';
+
+      if (!types.has(GLOSSARY_TERM_DRAG_TYPE)) {
+        operation = 'cancel';
+      } else if (target.type === 'item') {
+        const record = glossaryTermByFqn.get(String(target.key));
+        const isReparentTarget =
+          target.dropPosition === 'on' && !!record && !record.isLoadMoreButton;
+        operation = isReparentTarget ? 'move' : 'cancel';
+      }
```
[the remark this answers is not in the document]
Comment (siddhant1) [reply openmetadata-ui/src/main/resources/ui/src/components/Glossary/GlossaryTermTab/GlossaryTermTab.component.tsx:1614]: There's no sibling order

**Review by anuj-kumary — APPROVED** [review]
